src/main/java/net/dries007/tfc/client/screen/CharcoalForgeScreen.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.client.screen;
+
+import net.minecraft.entity.player.PlayerInventory;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.text.ITextComponent;
+
+import com.mojang.blaze3d.matrix.MatrixStack;
+import net.dries007.tfc.common.capabilities.heat.Heat;
+import net.dries007.tfc.common.container.CharcoalForgeContainer;
+import net.dries007.tfc.common.tileentity.CharcoalForgeTileEntity;
+
+import static net.dries007.tfc.TerraFirmaCraft.MOD_ID;
+
+public class CharcoalForgeScreen extends TileEntityScreen<CharcoalForgeTileEntity, CharcoalForgeContainer>
+{
+    private static final ResourceLocation FORGE = new ResourceLocation(MOD_ID, "textures/gui/charcoal_forge.png");
+
+    public CharcoalForgeScreen(CharcoalForgeContainer container, PlayerInventory playerInventory, ITextComponent name)
+    {
+        super(container, playerInventory, name, FORGE);
+        inventoryLabelY += 20;
+        imageHeight += 20;
+    }
+
+    @Override
+    protected void renderBg(MatrixStack matrixStack, float partialTicks, int mouseX, int mouseY)
+    {
+        super.renderBg(matrixStack, partialTicks, mouseX, mouseY);
+        int temp = (int) (51 * ((CharcoalForgeTileEntity) tile).getSyncableData().get(CharcoalForgeTileEntity.DATA_SLOT_TEMPERATURE) / Heat.maxVisibleTemperature());
```

Comment:
This cast shouldn't be necessary. Because the variable here:

https://github.com/TerraFirmaCraft/TerraFirmaCraft/blob/cc08a576ad03949220f89d78084ee4de0b3a5ad4/src/main/java/net/dries007/tfc/client/screen/TileEntityScreen.java#L18

is typed wrong. That should be `T`, which should then make this cast redundant.